src/module/token/metadata/BatchMetadataERC721.sol

```diff
@@ -153,6 +157,45 @@ contract BatchMetadataERC721 is Module, UpdateMetadataCallbackERC721 {
         return _batchMetadataStorage().nextTokenIdRangeStart;
     }
 
+    /// @dev Returns the id for the batch of tokens the given tokenId belongs to.
+    function getBatchId(uint256 _tokenId) public view virtual returns (uint256 batchId, uint256 index) {
+        uint256[] memory rangeEnds = _batchMetadataStorage().tokenIdRangeEnd;
+        uint256 numOfBatches = rangeEnds.length;
+
+        for (uint256 i = 0; i < numOfBatches; i += 1) {
+            if (_tokenId < rangeEnds[i]) {
+                index = i;
+                batchId = rangeEnds[i];
+
+                return (batchId, index);
+            }
+        }
+        revert BatchMetadataNoMetadataForTokenId();
+    }
+
+    /// @dev returns the starting tokenId of a given batchId.
+    function getBatchStartId(uint256 _batchID) public view returns (uint256) {
```

Comment:
i think what would be helpful for this, is to give me the token id range (start, end) for a given batch id

Reply:
updated